source/Engine.cpp

```diff
@@ -1588,9 +1588,12 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 	bool isJump = ship->IsUsingJumpDrive();
 	bool wasHere = (flagship && ship->GetSystem() == flagship->GetSystem());
 	bool wasHyperspacing = ship->IsHyperspacing();
+	bool wasDisabled = ship->IsDisabled();
 	// Give the ship the list of visuals so that it can draw explosions,
 	// ion sparks, jump drive flashes, etc.
 	ship->Move(newVisuals, newFlotsam);
+	if(ship->IsDisabled() && !wasDisabled)
+		eventQueue.emplace_back(nullptr, ship->shared_from_this(), ShipEvent::DISABLE);
```

Comment:
Collision detection does shared_from_this because collision detection is dealing with basic pointers to ships. This function is already using shared pointers. See below where it simply passes `ship`.
```suggestion
		eventQueue.emplace_back(nullptr, ship, ShipEvent::DISABLE);
```

Reply:
wooops, fixed